libkmod/libkmod-index.c

```diff
@@ -628,18 +628,14 @@ struct index_mm_value {
 	const char *value;
```

Comment:
We use the length as argument for `write_str_safe`. If we don't take the value which we already have here, we would have to call `strlen` again.

Reply:
I think we can live with that ... At worst we're moving it from the early parsing stage to the writing. We're not adding new strlen calls AFAICT.

Reply:
We must increment the pointer `p` though. By how much? It's what `strlen` tells us here. It can be skipped if we know that only one value is of interest, but we would duplicate this inline function for this one case. I'm really not sure if it's worth it, having two value parsing functions because of it. From a pure assembler point of view, it is.

Reply:
Can i offer some virtual :cookie: to give that a test and see how the LoC and perf look? Legibility should be fine IMHO although that's more subjective.